### PR TITLE
Propagate master only when analysis changes

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Alternative to #204 cc @art-w 
We just have to wait for https://github.com/ocurrent/ocurrent/pull/330 to be merged and this is good to go.

Already being tested in production.

### CODE REVIEW

**Comment by art-w** [comment]
Yes this is a much better fix! Thanks :)

File: service/pipeline.ml [hunk 1]
```diff
@@ -335,8 +335,9 @@ end
 let latch ~label x =
   let prev = ref (Error (`Active `Ready), None) in
   Current.component "latch %s" label |>
-  let> x = Current.state ~hidden:true x in
-  Result.iter (fun x -> prev := Ok x, None) x;
+  let> x = Current.state ~hidden:true x
+  and> meta = Current.Analysis.metadata x in
+  Result.iter (fun x -> prev := Ok x, meta) x;
```
Comment (kit-ty-kate) [inline service/pipeline.ml:340]: There seem to be some issues here. The metadata and the state aren't in sync which can cause problem:
scrn-2023-01-30-19-58-47
Here is the log (and previous working logs): https://opam-repo-ci.ci3.ocamllabs.io/job/2023-01-30/183948-opam-ci-analyse-9093d6

File: service/pipeline.ml [hunk 2]
```diff
@@ -363,12 +364,22 @@ let get_prs repo =
   in
   master, prs
 
+let analyze ~master src =
+  let master_analysis =
+    Analyse.examine ~master src
+    |> Current.cutoff (* keep the old master if the analysis hasn't changed *)
+       ~eq:(fun (_, old_analysis) (_, new_analysis) ->
+              Analyse.Analysis.equal old_analysis new_analysis)
```
[the remark this answers is not in the document]
Comment (kit-ty-kate) [reply service/pipeline.ml]: fixed

**Comment by kit-ty-kate** [comment]
Tested successfully (so far) live

**Pull request merged**